agent/bench-scripts/pbench-user-benchmark

```diff
@@ -30,209 +31,324 @@ benchmark="pbench-user-benchmark"
+		# We are not using tool triggers, so we'll only have one
+		# iteration, "1-default/${sample_name}/", so for compatibility
+		# we'll place the results output file there.
+		_final_results_dir="${benchmark_results_dir}"
+	else
+		# We are using tool triggers, so we'll potentially have more
+		# than one iteration, so just place the results output file in
+		# the run directory.
+		_final_results_dir="${benchmark_run_dir}"
+	fi
+	printf -- "%s%s | tee %s/result.txt" "${iter_benchmark_bin}" "${capture_stderr}" "${_final_results_dir}" > ${benchmark_run_cmd}
+	if [[ ${use_tool_triggers} -ne 0 ]]; then
+		printf -- " | pbench-tool-trigger ${iteration} ${benchmark_run_dir} ${tool_group}" >> ${benchmark_run_cmd}
+	fi
+	printf -- "\n" >> ${benchmark_run_cmd}
+	chmod +x ${benchmark_run_cmd}
 
-echo "Running $benchmark_bin"
-log "[$script_name] Running $benchmark_bin"
+	if [[ ${use_tool_triggers} -eq 0 ]]; then
+		if [[ ! -z "${pbench_pre}" ]]; then
+			log "[${script_name}]: Running ${pbench_pre} for iteration ${iteration_name}"
+			eval ${pbench_pre}
+			if [[ ${?} != 0 ]]; then
+				error_log "[${script_name}]: the --pbench-pre script, '${pbench_pre}', failed"
+				exit 1
+			fi
+		fi
+		pbench-start-tools --group=${tool_group} --iteration=${iteration} --dir=${benchmark_results_dir}
+	fi
 
-SECONDS=0
-$benchmark_run_cmd
-benchmark_duration=$SECONDS
-# make it predictable in the unittest environment
-if [[ $_PBENCH_BENCH_TESTS == 1 ]]; then
-	benchmark_duration=0
-fi
+	_msg="Running ${iter_benchmark_bin} for iteration ${iteration_name}"
+	echo "${_msg}"
+	log "[${script_name}] ${_msg}"
 
-if [ $use_tool_triggers -eq 0 ]; then
-	pbench-stop-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
-	pbench-postprocess-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
-	ln -s $(basename ${benchmark_results_dir}) $(dirname ${benchmark_results_dir})/reference-result
-else
-	result_dirs=$(ls -1d ${benchmark_run_dir}/*/sample1)
-	for rdir in $result_dirs; do
-		iteration_dir=$(basename $(dirname ${rdir}))
-		pbench-postprocess-tools --group=${tool_group} --iteration=${iteration_dir} --dir=${rdir}
-		ln -s $(basename ${rdir}) $(dirname ${rdir})/reference-result
-	done
-fi
-pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir end
+	SECONDS=0
+	${benchmark_run_cmd}
+	benchmark_duration=${SECONDS}
+	# make it predictable in the unittest environment
+	if [[ ${_PBENCH_BENCH_TESTS} == 1 ]]; then
+		benchmark_duration=0
+	fi
 
-if [[ ! -z $pbench_post ]]; then
-	log "[$script_name]: Running $pbench_post"
-	eval $pbench_post
-	if [[ $? != 0 ]]; then
-		error_log "[$script_name]: the script executed using pbench_post flag failed"
-		exit 1
+	if [[ $use_tool_triggers -eq 0 ]]; then
+		pbench-stop-tools --group="${tool_group}" --iteration="${iteration}" --dir="${benchmark_results_dir}"
+		pbench-postprocess-tools --group="${tool_group}" --iteration="${iteration}" --dir="${benchmark_results_dir}"
```

Comment:
pbench-start-tools and pbench-stop-tools are pointing towards pbench-postprocess-tools only. so, how are they differ from each other?

Reply:
@riya-17, as you can see in `pbench-postprocess-tools` line [12](https://github.com/distributed-system-analysis/pbench/blob/master/agent/util-scripts/pbench-postprocess-tools#L12), it determines what "action" to take based on the symlink and/or name of the script.

Reply:
Yaa got it.. Thanks @portante